analyses/cell-type-nonETP-ALL-03/environment.yml

```
@@ -0,0 +1,20 @@
+name: "openscpca-cell-type-nonETP-ALL-03"
+channels:
+  - conda-forge
+  - bioconda
+platforms:
+  - linux-64
+  - osx-64
+  - osx-arm64
+dependencies:
+  - python=3.11
+  - awscli>=2.15
+  - conda-lock>=2.5
+  - jq>=1.7
+  - pre-commit>=3.6
+  - session-info>=1.0
+  - anndata=0.10.9
+  - leidenalg=0.10.2
+  - pip
+  - pip:
+      - sam-algorithm==1.0.2
```

When I tried to install the environment using `conda-lock install` I ran into an error with during `pip` installation part of the process. This has happened to me in the past, and the usual solution is to try to move a few more dependencies into the conda-managed section of the environment. I also recommend generally leaving third decimals off of the `environment.yaml` file; they will be captured by `conda-lock` either way but allow a bit more flexibility in initial solving of the environment. 

The suggestion below moves the dependencies that I could out of `pip` (this also implicitly includes the `scikit-learn` and `numba` packages, which seem most likely to cause installation errors).

```suggestion
  - anndata=0.10
  - leidenalg=0.10
  - dill
  - harmonypy
  - umap-learn
  - pip
  - pip:
      - sam-algorithm==1.0.2
```

After updating this file, I was able to run `conda-lock --update '' ` (The quoted empty string forces re-resolving the entire environment) which resulted in a `conda-lock.yml` that installed without trouble, at least at my end.

---

I made the suggested changes, and ran `conda-lock --update ''`. Then I created a new conda environment with `conda-lock install --name tmp conda-lock.yml`, but it fails to import samalg. Did you manage to import samalg in python?

---

Yes, I think that should be fine. I will do some further testing to see if I can figure out why it is not being reliable, but we do not need to have that hold things up for now.

---

I did a bit more testing and pushed https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/767/commits/353bcb75fb9574f690e4ca5cec427da040965c05 which removed some of the `pip` installed packages from the conda-lock file that may have been causing trouble. (I didn't do this manually, but by deleting the `conda-lock.yml` file and regenerating it, but the diff is quite clean). 

If you have a chance to test `conda-lock install` with this file, it would be great to know if that solved the problem.

---

Yes, I managed to install the conda environment perfectly, and I can import samalg. Thank you so much for your help! 
Just to clarify, which of the `pip` installed packages did you remove from the conda-lock file? I am wondering how could I generate this conda-lock file that works, if I happen to update the `environment.yml` in the future and thus need to change the `conda-lock.yml`?